src/core_init_atmosphere/mpas_init_atm_gwd.F

```diff
@@ -305,48 +334,231 @@ function compute_gwd_fields(domain) result(iErr)
+      
+      implicit none
+
+      real (kind=RKIND), intent(in) :: lat
+      real (kind=RKIND), intent(in) :: lon
+      real (kind=RKIND), intent(in) :: dx
+
+      integer, intent(out) :: nx
+      integer, intent(out) :: ny
+
+      !
+      ! Get number of points to extract in the zonal direction
+      !
+      if (cos(lat/rad2deg) > (2.0 * pts_per_degree * dx * 180.0) / (real(topo_x,RKIND) * Pi * Re)) then
+         nx = ceiling((180.0 * dx * pts_per_degree) / (Pi * Re * cos(lat/rad2deg)))
+      else
+         nx = topo_x / 2
+      end if
+
+      !
+      ! Get number of points to extract in the meridional direction
+      !
+      ny = ceiling((180.0 * dx * pts_per_degree) / (Pi * Re))
+
+
+   end subroutine get_box_size_from_lat_lon
+
+
+
+   !***********************************************************************
+   !
+   !  function get_tile_from_box_point
+   !
+   !> \brief   Routine to obtain a tile given a box pixel
+   !> \author  Abishek Gopal
+   !> \date    05 Sep 2024
+   !> \details 
+   !>  Routine to obtain a tile of type mpas_gwd_tile_type, given the linked 
+   !    list of tiles tilesHead, box coordinates box_x, box_y, and path to 
+   !    static dataset. The function first searches the linked list to locate
+   !    the tile, and if search fails, adds the target tile to the linked list
+   !    after reading in the data for the tile from disk 
+   !-----------------------------------------------------------------------
+   function get_tile_from_box_point(tilesHead, box_x, box_y, path, sub_path) result(thisTile)
+
+      implicit none
+      integer, intent(in) :: box_x, box_y
+      type(mpas_gwd_tile_type), pointer, intent(in) :: tilesHead
+      type(mpas_gwd_tile_type), pointer :: thisTile
+      character(len=*), intent(in) :: path
+      character(len=*), intent(in) :: sub_path
+
+      integer :: tile_start_x, tile_start_y, tile_start_x_topo
+      
+      ! Need special handling for the x-coordinates of topo tiles, due to the shift by topo_shift
+      ! in certain datasets. We use tile_start_x, tile_start_y to search for tiles and open landmask tiles,
+      ! whereas tile_start_x_topo is only required to open the correct topo tiles from disk
+      if (box_x > topo_shift) then
+         tile_start_x_topo = floor( real(box_x - topo_shift - 1) / real(tile_x)) * tile_x + 1
+      else
+         tile_start_x_topo = floor( real(box_x + topo_shift - 1) / real(tile_x)) * tile_x + 1
+      end if
+      tile_start_x = floor( real(box_x - 1) / real(tile_x)) * tile_x + 1
+      tile_start_y = floor( real(box_y - 1) / real(tile_y)) * tile_y + 1
+
+
+      
+      thisTile => tilesHead
+      ! loop over tiles
+      do while (associated(thisTile))
+
+         if(thisTile%tile_start_x==tile_start_x .and. thisTile%tile_start_y==tile_start_y) then 
+            exit
+         end if
+
+         thisTile => thisTile % next
+
+      enddo   ! associated(thisTile)
```

Comment:
Just a note about coding style: throughout MPAS we tend towards the use of `end if` and `end do` with a space.

Reply:
thanks! I didn't notice it.